replica/table.cc

```diff
@@ -2350,6 +2355,16 @@ table::table(schema_ptr schema, config config, lw_shared_ptr<const storage_optio
     set_metrics();
 }
 
+void table::on_flush_timer() {
+    tlogger.debug("on table {}.{} flush timer, period {}ms", _schema->ks_name(), _schema->cf_name(), _schema->memtable_flush_period());
+    (void)with_gate(_async_gate, [this] {
+        return flush().then([this] {
+            _flush_timer.arm(std::chrono::milliseconds(_schema->memtable_flush_period()));
+            return make_ready_future<>();
+        });
+    });
```

Comment:
Better to use finally() to re-arm even on failure.

Reply:
This is susceptible to a race - if the schema is changed while a flush is ongoing, you don't notice it.

Reply:
> This is susceptible to a race - if the schema is changed while a flush is ongoing, you don't notice it.

Yes, I thought about it. But what will happen in current implementation if we change schema at the moment of flushing? Seems it's possible

Reply:
> Better to use finally() to re-arm even on failure.

Probably it would be even better to catch and log an exception and then re-arm timer?